api/api-base/src/test/groovy/com/thoughtworks/go/api/DeprecationHeadersTrait.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.thoughtworks.go.api
+
+import com.google.gson.Gson
+import com.thoughtworks.go.api.mocks.MockHttpServletResponseAssert
+import com.thoughtworks.go.api.util.HaltApiMessages
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.invocation.InvocationOnMock
+import spark.Request
+import spark.Response
+
+import static org.mockito.ArgumentMatchers.any
+import static org.mockito.Mockito.doAnswer
+
+trait DeprecationHeadersTrait {
+  abstract void makeHttpCall()
+
+  abstract String getControllerMethodUnderTest()
+
+  @BeforeEach
+  void stubControllerAction() {
+    doAnswer({ InvocationOnMock invocation -> return new Gson().toJson([message: "response"]) })
+      .when(controller)."${controllerMethodUnderTest}"(any() as Request, any() as Response)
+  }
+
+  @Test
+  void 'should verify deprecation headers'() {
+    makeHttpCall()
+
+    if (controller.isDeprecated()) {
+      verifyDeprecationHeadersArePresent()
+    } else {
+      verifyNoDeprecationHeaders()
+    }
+  }
+
+  def verifyDeprecationHeadersArePresent() {
+    ((MockHttpServletResponseAssert) assertThatResponse())
+      .hasHeader("X-GoCD-API-Deprecation-Version", "v${controller.deprecatedRelease}")
+      .hasHeader("X-GoCD-API-Removal-Version", "v${controller.removalRelease}")
+      .hasHeader("X-GoCD-API-Deprecation-Info", "https://api.gocd.org/${controller.deprecatedRelease}/#api-changelog")
+      .hasHeader("Link", "<${request.getRequestURL().toString()}>; Accept=\"${controller.successorVersion.mimeType}\"; rel=\"successor-version\"")
```

Comment:
```
X-API-Deprecated-In: GoCD v20.2.0
X-API-Removal-In: GoCD v20.5.0
X-API-Deprecation-Info: "https://api.gocd.org/20.2.0/#api-changelog"
```
WDYT?

Reply:
Hmm, my preference was to keep the `GoCD` in the header key, as it makes it explicit that it is a GoCD specific header.